src/shared/components/upload/helpers/uploadHelpers.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import windowService from 'src/services/window-service';
+
+import type {
+  Options,
+  Result,
+  OptionsWithDefinedTransform,
+  TransformTo
+} from '../types';
+
+// The File API also includes the `readAsBinaryString` method which is not recommended
+// (readAsArrayBuffer should be used instead)
+// (see: https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsBinaryString)
+type FileReaderMethod = 'readAsArrayBuffer' | 'readAsDataURL' | 'readAsText';
+
+const transformToFileReaderMethodMap: Record<TransformTo, FileReaderMethod> = {
+  arrayBuffer: 'readAsArrayBuffer',
+  dataUrl: 'readAsDataURL',
+  text: 'readAsText'
+};
+
+export const transformFiles = async <
+  T extends Options & OptionsWithDefinedTransform
+>(
+  files: File[],
+  options: T
+): Promise<Result<T>> => {
+  const { allowMultiple, transformTo } = options;
```

Comment:
Why should transformFiles worry about `allowMultiple` instead of just transforming the files it gets?

Reply:
Oof. Fair point. I think I did it in order to make the types work. I'll see if typescript will object if I just pass a file or an array of files.

Reply:
Updated the signature of `transformFiles`. It didn't upset typescript and simplified the code.